api.js

```diff
@@ -192,11 +192,7 @@ Api.prototype.run = function (files) {
 			self.fileCount = files.length;
 			self.base = path.relative('.', commonPathPrefix(files)) + path.sep;
 
-			var tests = files.map(self._runFile);
-
-			// receive test count from all files and then run the tests
-			var unreportedFiles = self.fileCount;
-
+			var tests = new Array(self.fileCount);
```

Comment:
It gets filled with promises for run results [here](https://github.com/sindresorhus/ava/blob/watch-and-syntax-errors/api.js#L248)